cpp/src/toppra/constraint.cpp

```diff
@@ -42,6 +47,10 @@ void allocateLinearPart(std::size_t N, Eigen::Index k, Eigen::Index m,
   for (Vector& x : g) x.resize(k);
 }
 
+void checkSizes (std::size_t N, Eigen::Index k, Eigen::Index m,
```

Comment:
Why are there two definitions of this function?

Reply:
There is one declaration and one definition. Visibility flag can only be put in the declaration.

Reply:
You may have a look at https://www.tutorialspoint.com/What-is-the-difference-between-a-definition-and-a-declaration-in-Cplusplus

Reply:
I see. So instead of putting the declaration in the header file, it is placed directly to the source file. This is possible because no other classes make use of this function anyway. 

Is it valid to put TOPPRA_NO_EXPORT to private class methods in header files?

Reply:
Yes, it is valid. I don't know whether it is useful because I don't what is the default for private class method.

The export stuff is generally required when supporting several compilers because all the compilers do not use the same default visibility. VC++ hides symbols by default while gcc makes them visible by default...